src/views/CountryRiskWatch/RiskBarChart/FoodInsecurityChart/index.tsx

```diff
@@ -0,0 +1,294 @@
+    navigator.language,
+    { month: 'short' },
+);
+
+const formatDate = (date: Date) => date.toLocaleString(
+    navigator.language,
+    {
+        year: 'numeric',
+        month: 'short',
+    },
+);
+
+interface Props {
+    ipcData: RiskData['ipc_displacement_data'] | undefined;
+}
+
+function FoodInsecurityChart(props: Props) {
+    const { ipcData } = props;
+
+    const chartContainerRef = useRef<HTMLDivElement>(null);
+    const chartBounds = useSizeTracking(chartContainerRef);
+
+    const chartData = useMemo(
+        () => {
+            const uniqueData = getPrioritizedIpcData(ipcData ?? []);
+            const averageDataItem = getAverageIpcData(uniqueData);
+            const maxValue = maxSafe(uniqueData.map((item) => item.total_displacement)) ?? 0;
+
+            const xScale = getScaleFunction(
+                { min: 0, max: 12 },
+                { min: 0, max: chartBounds.width },
+                { start: chartMargin.left, end: chartMargin.right },
+            );
+
+            const yScale = getScaleFunction(
+                { min: 0, max: maxValue },
+                { min: 0, max: chartBounds.height },
+                { start: chartMargin.top, end: chartMargin.bottom },
+                true,
+            );
+
+            const yearGroupedData = mapToMap(
+                listToGroupList(
+                    uniqueData,
+                    (item) => item.year,
+                ),
+                (year) => year,
+                (ipcDataList) => getAverageIpcData(ipcDataList),
+            );
+
+            const yearKeys = Object.keys(yearGroupedData);
+            const currentYear = new Date().getFullYear();
+            const predictionYear = yearKeys[yearKeys.length - 1];
+
+            const chartPoints = Array.from(Array(12).keys()).map(
+                (monthKey) => {
+                    const month = monthNumberToNameMap[monthKey];
+                    const average = averageDataItem[month] ?? 0;
+
+                    const value = listToMap(
+                        yearKeys,
+                        (year) => year,
+                        (year) => yearGroupedData?.[Number(year)]?.[month],
+                    );
+
+                    value.average = average;
+
+                    const y = mapToMap(
+                        value,
+                        (key) => key,
+                        (item) => (isDefined(item) ? yScale(item) : undefined),
+                    );
+
+                    const date = new Date(currentYear, monthKey, 1);
+
+                    return {
+                        key: monthKey,
+                        x: xScale(monthKey),
+                        y,
+                        value,
+                        date,
+                    };
+                },
+            );
+
+            const numYAxisPoints = 6;
+            const diff = maxValue / (numYAxisPoints - 1);
+            const yAxisPoints = Array.from(Array(numYAxisPoints).keys()).map(
```

Comment:
Let's check if we can use Array.fill

Reply:
No, `Array.fill` is only used when we need same value filled throughout the array